apps/agreement/contracts/Agreement.sol

```diff
@@ -110,238 +108,176 @@ contract Agreement is IArbitrable, IForwarder, AragonApp {
     */
-    function receiveApproval(address _from, uint256 _amount, address _token, bytes /* _data */) external onlySigner(_from) {
-        require(msg.sender == _token && _token == address(collateralToken), ERROR_SENDER_NOT_ALLOWED);
-        _stakeBalance(_from, _from, _amount);
-    }
+    function newAction(uint256 _disputableId, address _submitter, bytes _context) external returns (uint256) {
+        uint256 lastContentIdSigned = lastContentSignedBy[_submitter];
+        require(lastContentIdSigned >= _getCurrentContentId(), ERROR_SIGNER_MUST_SIGN);
 
-    /**
-    * @notice Unstake `@tokenAmount(self.collateralToken(): address, _amount)` tokens from `msg.sender`
-    * @param _amount Number of collateral tokens to be unstaked
-    */
-    function unstake(uint256 _amount) external {
-        require(_amount > 0, ERROR_INVALID_UNSTAKE_AMOUNT);
-        _unstakeBalance(msg.sender, _amount);
-    }
+        DisputableInfo storage disputableInfo = disputableInfos[msg.sender];
+        require(disputableInfo.state == DisputableState.Registered, ERROR_DISPUTABLE_APP_NOT_REGISTERED);
 
-    /**
-    * @notice Schedule a new action
-    * @param _context Link to a human-readable text giving context for the given action
-    * @param _script Action script to be executed
-    */
-    function schedule(bytes _context, bytes _script) external onlySigner(msg.sender) {
-        _createAction(msg.sender, _context, _script);
-    }
+        uint256 currentCollateralRequirementId = _getCurrentCollateralRequirementId(disputableInfo);
+        CollateralRequirement storage requirement = disputableInfo.collateralRequirements[currentCollateralRequirementId];
+        _lockBalance(requirement.token, _submitter, requirement.actionAmount);
 
-    /**
-    * @notice Execute action #`_actionId`
-    * @dev It only executes non-challenged actions after the challenge period or actions that were disputed but ruled in favor of the submitter
-    * @param _actionId Identification number of the action to be executed
-    */
-    function execute(uint256 _actionId) external {
-        (Action storage action, Setting storage setting) = _getActionAndSetting(_actionId);
-        require(_canExecute(action), ERROR_CANNOT_EXECUTE_ACTION);
+        uint256 id = actions.length++;
+        Action storage action = actions[id];
+        action.disputable = IDisputable(msg.sender);
+        action.collateralId = currentCollateralRequirementId;
+        action.disputableId = _disputableId;
+        action.submitter = _submitter;
+        action.context = _context;
 
-        if (action.state == ActionState.Scheduled) {
-            _unlockBalance(action.submitter, setting.collateralAmount);
-        }
-        action.state = ActionState.Executed;
-        runScript(action.script, new bytes(0), _getScriptExecutionBlacklist());
-        emit ActionExecuted(_actionId);
+        disputableInfo.ongoingActions++;
+        emit ActionSubmitted(id);
```

Comment:
The rationale here, since the Agreement's size is almost hitting the limit (I already had to reduce the number of runs) was to emit the less amount of info we can while making sure it could be retrieved afterward. In this case, I made sure all the actions-related information could be fetched using their IDs. Happy to discuss it tho :)